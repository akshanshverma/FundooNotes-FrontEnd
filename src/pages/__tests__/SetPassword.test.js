import React from 'react';
import { shallow } from 'enzyme';
import SetPassword from "../SetPassword";


const component = shallow(<SetPassword debug />);
describe('SetPassword', () => {
    it('should render correctly in "debug" mode SetPassword', () => {
        expect(component).toMatchSnapshot();
    });


    it('should change event and password set in state for reset', () => {
        const emailInput = component.find('#resetPassword');
        emailInput.props().onChange({
            target: {
                name: 'password',
                value: 'qwerty'
            }
        });
        expect(component.state('password')).toEqual('qwerty');
    });


    it('should change event and rpassword in state to check confirm password', () => {
        const emailInput = component.find('#resetRPassword');
        emailInput.props().onChange({
            target: {
                name: 'rpassword',
                value: 'qwerty'
            }
        });
        expect(component.state('rpassword')).toEqual('qwerty');
    });
});