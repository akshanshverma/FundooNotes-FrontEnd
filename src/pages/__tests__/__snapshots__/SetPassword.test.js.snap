// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetPassword should render correctly in "debug" mode SetPassword 1`] = `
<div
  className="maindiv"
>
  <WithStyles(Card)
    id="card"
  >
    <div>
      <WithStyles(Typography)
        color="primary"
        id="registerT"
      >
        Set Password
      </WithStyles(Typography)>
      <div
        className="hold"
      >
        <div
          className="input"
        >
          <Input
            id="resetPassword"
            label="password"
            name="password"
            onChange={[Function]}
            placeholder="enter password"
            type="password"
          />
          <div
            className="msg"
          />
        </div>
        <div
          className="input"
        >
          <Input
            id="resetRPassword"
            label="confirm password"
            name="rpassword"
            onChange={[Function]}
            placeholder="confirm password"
            type="password"
          />
          <div
            className="msg"
          />
        </div>
        <div
          className="button"
        >
          <WithStyles(Button)
            color="primary"
            id="saveButton"
            onClick={[Function]}
            variant="contained"
          >
            save
          </WithStyles(Button)>
        </div>
        <div
          className="errMsg"
        />
      </div>
    </div>
  </WithStyles(Card)>
</div>
`;
